fedimint-core/src/db/mod.rs

```diff
@@ -206,29 +216,29 @@ impl Database {
                         Ok(()) => {
                             return Ok(val);
                         }
-                        Err(e) if max_retries.map(|mr| mr <= retries).unwrap_or(false) => {
```

Comment:
Previous version was shorter and better, IMO. :P

Reply:
:D It was definitely shorter. It just took me a couple minutes to wrap my head around the line every time I looked at it. I felt that it's kinda like the double-negative example we had with the environment variable for testing. 

I can revert that if you like :)

Reply:
> I felt that it's kinda like the double-negative example we had with the environment variable for testing.

But ... there's no negation here, no? "max_retries <= retries if Some, otherwise false"

What I don't like is, personally, is just using the `match x { pattern if expression => ... }` thing altogether. I'd merge the `Err` case and make it a normal:

```
Err(e) => if max_retries.map(|mr| mr <= retries).unwrap_or(false) { 
   // try again
   } else {
      return Err(... stuff... );
   }
```

Ability to add condition to pattern is sometimes nice, but oftentimes it's just extra idiosyncrasy.

Reply:
> I can revert that if you like :)

We should let the next reviewer be a tie-breaker. :D

Reply:
hm, right, 
that 
```rust
Err(e) => if max_retries.map(|mr| mr <= retries).unwrap_or(false) { 
   // try again
   } else {
      return Err(... stuff... );
   }
```
would work as well and is a bit more concise which I like.

If nobody steps up with an opinion until tomorrow when I look at this again, I'll change it to that, and we can merge it. I don't have a strong opinion on this and the change is about the `max_attempts` thing anyway :D

Reply:
I vote for the old way or just make max_retries `u64` and default to MAX, so you can remove that comment too.

Reply:
ok, thanks @jkitman. reverted to more or less the old way that also removes the comment